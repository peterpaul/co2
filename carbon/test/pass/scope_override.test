["stdlib.h"] {
	int printf(char *format, ...);
}

int main () {
	B bar = new B();
	bar.foo = 5;
	bar.output (5);
	return 0;
}

class B : A {
	int bar (int baz) {
		return super(baz) + 5;
	}

	void output (int a) {
		printf ("foo = %d, bar (%d) = %d\n", foo, a, bar (a));
	}
}

class A {
	int foo;
	int bar (int baz) {
		return foo + baz;
	}
}

