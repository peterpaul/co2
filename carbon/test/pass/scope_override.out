foo = 5, bar (5) = 15
